src/scratchpad.js

```diff
@@ -122,47 +124,173 @@ export default function Scratchpad (DOM, props) {
       sinks.dispose();
     }
 
-    const context = {error$, require, console};
+    // Store npm modules that have been loaded from http://wrzd.in.
+    let moduleCache = {};
 
-    const wrappedCode = `
-      try {
-        ${code}
+    // Matches for all npm modules that will need to be loaded.
+    const requireMatches = code.match(/require\(.*/g);
+
+    function runInVm (code) {
+      const context = {
+        error$: error$,
+        console: console,
+        require: (path) => {
+          return moduleCache[path];
+        }
+      };
+
+      const wrappedCode = `
+        try {
+          ${code}
+
+          error$.onNext('');
+        } catch (e) {
+          error$.onNext(e);
+        }
+      `;
 
-        error$.shamefullySendNext('');
+      try {
+        vm.runInNewContext(wrappedCode, context);
       } catch (e) {
         error$.shamefullySendNext(e);
       }
-    `;
 
-    try {
-      vm.runInNewContext(wrappedCode, context);
-    } catch (e) {
-      error$.shamefullySendNext(e);
-    }
+      if (typeof context.main !== 'function' || typeof context.sources !== 'object') {
+        return;
+      }
 
-    if (typeof context.main !== 'function' || typeof context.sources !== 'object') {
-      return;
-    }
+      let userApp;
 
-    let userApp;
+      if (!drivers) {
+        drivers = context.sources;
+      }
 
-    if (!drivers) {
-      drivers = context.sources;
-    }
+      try {
+        if (sources && restartEnabled) {
+          userApp = restart(context.main, drivers, {sources, sinks})
+        } else {
+          userApp = run(context.main, context.sources);
+        }
+      } catch (e) {
+        error$.onNext(e);
+      }
 
-    try {
-      if (sources && restartEnabled) {
-        // userApp = restart(context.main, drivers, {sources, sinks})
-      } else {
-        userApp = run(context.main, context.sources);
+      if (userApp) {
+        sources = userApp.sources;
+        sinks = userApp.sinks;
       }
-    } catch (e) {
-      error$.shamefullySendNext(e);
     }
 
-    if (userApp) {
-      sources = userApp.sources;
-      sinks = userApp.sinks;
+    function fetchModules (moduleList) {
+      return new Promise((resolve, reject) => {
+        let dependencies = moduleList.reduce((acc, moduleName) => {
+          const moduleNameWithScope = moduleName.replace('/', '%2F');
+          acc[moduleNameWithScope] = 'latest';
+
+          return acc;
+        }, {});
+
+        let postData = {
+          'options': {
+            'debug': false,
+            'standalone': true
+          },
+          'dependencies': dependencies
+        };
+
+        const options = {
+          hostname: 'wzrd.in',
+          path: '/multi',
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json'
+          }
+        };
+
+        const req = https.request(options, (res) => {
```

Comment:
I would prefer to make http requests using `@cycle/http` if possible, to shift side effects out of the `main`.